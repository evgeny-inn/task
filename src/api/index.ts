export { default as client } from './httpClient';
