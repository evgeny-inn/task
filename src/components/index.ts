export { default as Header } from './Header/Header';
export { default as Main } from './Main/Main';
export { default as AddForm } from './AddForm/AddForm';
export { default as UpdateForm } from './UpdateForm/UpdateForm';
