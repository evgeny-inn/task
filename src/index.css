:root {
  --color-light: #FFF;
  --color-semi-light: #EBECEC;
  --color-semi-dark: #707070;
  --color-dark: #434242;
  --color-accent: #008A64;
  --color-danger: #FC2020;
}

html, body {
  height: 100%;
  margin: 0;
  color: var(--color-semi-dark);
  font-weight: 400;
  font-size: 16px;
  line-height: 21px;
  font-family: Roboto, sans-serif;
}

*, ::after, ::before {
  box-sizing: border-box;
}

:global(#root) {
  height: 100%;
  display: flex;
  flex-direction: column;
}
