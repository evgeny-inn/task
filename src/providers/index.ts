export { default as ContactsProvider, ContactsContext } from './ContactsProvider';
