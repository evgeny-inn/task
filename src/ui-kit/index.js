export { default as Button } from './Button/Button.jsx';
export { default as Input } from './Input/Input.jsx';
export { default as Record } from './Record/Record.jsx';
