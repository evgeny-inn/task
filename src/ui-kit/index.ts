export { default as Button } from './Button/Button';
export { default as Input } from './Input/Input';
export { default as Record } from './Record/Record';
export { default as Logo } from './Logo/Logo';
export { default as Modal } from './Modal/Modal';
