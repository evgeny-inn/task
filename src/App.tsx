import { ContactsProvider } from './providers';
import { Header, Main } from './components';
import Home from './pages/Home';
import './index.css';

const App = () => (
  <ContactsProvider>
    <Header />
    <Main>
      <Home />
    </Main>
  </ContactsProvider>
);

export default App;
