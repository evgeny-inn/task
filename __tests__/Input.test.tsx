import React from 'react';
import { Input } from '../src/ui-kit';
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<Input label="Label" />).toJSON();
  expect(tree).toMatchSnapshot();
});
