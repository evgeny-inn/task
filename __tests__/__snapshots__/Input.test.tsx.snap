// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
[
  <label
    htmlFor=":r0:"
  >
    Label
  </label>,
  <input
    id=":r0:"
  />,
]
`;
